app.js

```diff
@@ -56,12 +56,15 @@ function runNpm(command) {
 	process.exit(0);
 }
 
+var isLegacyEngine = !global.Map;
+
 try {
 	require('sugar');
+	if (isLegacyEngine) require('es6-shim');
 } catch (e) {
 	runNpm('install --production');
 }
-if (!Object.select) {
+if (isLegacyEngine && !(new Map().set(0, 0) instanceof Map)) {
```

Comment:
Not currently as the old <code>es6-shim</code> version fails this test, but the tournaments code would benefit in several places from chaining <code>Map#set</code<